### DISCUSSION TITLE
How to include or exclude fields for list of nested models?

### DISCUSSION THREAD

**Discussion by SDAravind** [pydantic/pydantic#9382] [Question] [answered]
I would like to include or exclude on list of nested models. How would I do that with below example? Please point me in right direction that would help me solve this.

```python
from pydantic import BaseModel


class Foo(BaseModel):
    a: str
    b: int


class Bar(BaseModel):
    c: str
    d: Foo
    e: list[Foo]


Bar(
    c="test", d=Foo(a="foo", b=10), e=[Foo(a="test1", b=1), Foo(a="test2", b=2)]
).model_dump(include={"c": True, "d": {"b"}, "e": {"a"}})
)
>>> {'c': 'test', 'd': {'b': 10}, 'e': []}
```
Expected Output: I would like see below output
```python
>>> {'c': 'test', 'd': {'b': 10}, 'e': [{'a': 'test1'}, {'a': 'test2'}]}

**Comment by xtendch** [comment]
Does https://github.com/pydantic/pydantic/discussions/6699 provide any answers?

**Reply by SDAravind** [reply]
Nope.

**Comment by AlessandroMiola** [comment, marked as answer]
Hi @SDAravind, 
the following should do the trick:

    bar.model_dump(include={"c": True, "d": {"b"}, "e": {"__all__": {"a"}}})

Quoting the [docs](https://docs.pydantic.dev/latest/concepts/serialization/#advanced-include-and-exclude)

> Special care must be taken when including or excluding fields from a list or tuple of submodels or dictionaries. In this scenario, model_dump and related methods expect integer keys for element-wise inclusion or exclusion. **To exclude** (ed. or include) **a field from every member of a list or tuple, the dictionary key** '\_\_all__' **can be used**, as shown here
